qadence_libs/qinfo_tools/spsa.py

```diff
@@ -0,0 +1,116 @@
+from __future__ import annotations
+
+import numpy as np
+import torch
+from qadence import Overlap
+from torch import Tensor
+
+
+def _create_random_direction(size: int) -> Tensor:
+    """
+    Creates a torch Tensor with elements randomly drawn from {-1,+1}.
+
+    Args:
+        size (int): Size of the vector
+    """
+    return torch.Tensor(np.random.choice([-1, 1], size=(size, 1)))
+
+
+def _shifted_overlap(
+    model: Overlap,
+    shift: Tensor,
+    fm_dict: dict,
+    vparams_dict: dict,
+) -> Tensor:
+    """
+    Forward method of the Overlap model with shifted values of the ket variational parameters.
+
+    Args:
+        model (Overlap): Overlap model
+        shift (float): Quantity of the shift
+        fm_dict (dict): Feature map dictionary
+        vparams_dict (dict): Variational parameter dictionary
+    """
+    shifted_vparams_dict = {k: (v + s) for (k, v), s in zip(vparams_dict.items(), shift)}
+
+    ovrlp_shifted = model(
+        bra_param_values=fm_dict | vparams_dict,
```

Comment:
What does the `|` do here ?

Reply:
It [merges](https://peps.python.org/pep-0584/) the two dictionaries